### PR TITLE
Support 'selector' in dynamic watchlists.

### PR DESCRIPTION
**Pull request by IvanIvanoff**
#### Summary
Rework some watchlist internals to support :ok/:error tuples
Support dynamic watchlists with `selector` function. It accepts the same arguments as the selector in the `allProjects` queries - `filters`, `orderBy` and `pagination`. Check the description of [this PR](https://github.com/santiment/sanbase2/pull/1936) for all supported fields.
Example:

```graphql
    mutation {
      createWatchlist(
        name: \"My list\"
        color: BLACK
        function: \"{\"args\":{\"filters\":[{\"aggregation\":\"last\",\"from\":\"2020-06-09 11:08:42.006876Z\",\"metric\":\"daily_active_addresses\",\"operator\":\"greater_than_or_equal_to\",\"threshold\":10,\"to\":\"2020-06-16 11:08:42.016549Z\"}]},\"name\":\"selector\"}\"
        ) {
         id
         name
         color
         isPublic
         user{ id }

         listItems{
           project{ slug }
         }
      }
    }
```

[//]: # (#### Related PRs)

[//]: # (#### Additional deploy notes)

[//]: # (#### Screenshots)

### CODE REVIEW

File: lib/sanbase/user_lists/watchlist_function.ex [hunk 1]
```diff
@@ -8,13 +8,24 @@ defmodule Sanbase.WatchlistFunction do
...
   def type, do: :map
 
+  def evaluate(%__MODULE__{name: "selector", args: args}) do
+    case Map.split(args, ["filters", "order", "pagination"]) do
+      {selector, empty_map} when map_size(empty_map) == 0 ->
+        {:ok, projects} = Project.ListSelector.projects(%{selector: selector})
+        projects
+
+      {_selector, unsupported_keys_map} ->
+        {:error}
```
Comment (tspenov) [inline lib/sanbase/user_lists/watchlist_function.ex:18]: Missing error message.

Comment (IvanIvanoff) [reply lib/sanbase/user_lists/watchlist_function.ex]: fixed

File: lib/sanbase/model/project/list/list_selector.ex [hunk 2]
```diff
@@ -0,0 +1,84 @@
+defmodule Sanbase.Model.Project.ListSelector do
```
Comment (tspenov) [inline lib/sanbase/model/project/list/list_selector.ex:1]: We can reuse some of this code here: https://github.com/santiment/sanbase2/blob/980f4d627af1b6cf27f92217dd77a765cd715298/lib/sanbase_web/graphql/resolvers/project/project_list_resolver.ex

Comment (IvanIvanoff) [reply lib/sanbase/model/project/list/list_selector.ex:1]: Yeah, that was the idea - I copy-pasted it but forgot to change it there.

**Pull request merged**